langstream-agents/langstream-agent-grpc/src/main/java/ai/langstream/agents/grpc/GrpcAgentProcessor.java

```diff
@@ -77,30 +80,41 @@ public GrpcAgentProcessor(ManagedChannel channel) {
     }
 
     @Override
-    public void start() {
+    public void start() throws Exception {
         if (channel == null) {
             throw new IllegalStateException("Channel not initialized");
         }
-        request = ProcessorGrpc.newStub(channel).process(responseObserver);
+        blockingStub = AgentServiceGrpc.newBlockingStub(channel);
+        request = AgentServiceGrpc.newStub(channel).withWaitForReady().process(responseObserver);
     }
 
     @Override
-    public void setContext(AgentContext context) {
+    public void setContext(AgentContext context) throws Exception {
         this.agentContext = context;
     }
 
+    @Override
+    protected Map<String, Object> buildAdditionalInfo() {
+        try {
+            return MAPPER.readValue(
+                    blockingStub.agentInfo(Empty.getDefaultInstance()).getJsonInfo(), Map.class);
```

Comment:
is there a timeout here or are we waiting forever ?

Reply:
done